iotedgedev/edge.py

```diff
@@ -1,16 +1,37 @@
+from iotedgedev.output import Output
+from iotedgedev.envvars import EnvVars
+from iotedgedev.azurecli import AzureCli
+
+
 class Edge:
-    def __init__(self, envvars, output, azure_cli):
+    def __init__(self, envvars: EnvVars, output: Output, azure_cli: AzureCli):
         self.envvars = envvars
         self.output = output
         self.azure_cli = azure_cli
 
-    def deploy(self, manifest_file):
+    def deploy(self, manifest_file: str):
 
         self.output.header("DEPLOYING CONFIGURATION")
 
         self.envvars.verify_envvar_has_val("IOTHUB_CONNECTION_STRING", self.envvars.IOTHUB_CONNECTION_INFO)
         self.envvars.verify_envvar_has_val("DEVICE_CONNECTION_STRING", self.envvars.DEVICE_CONNECTION_INFO)
         self.envvars.verify_envvar_has_val("DEPLOYMENT_CONFIG_FILE", self.envvars.DEPLOYMENT_CONFIG_FILE)
 
-        if self.azure_cli.set_modules(self.envvars.DEVICE_CONNECTION_INFO.device_id, self.envvars.IOTHUB_CONNECTION_INFO, manifest_file):
+        if self.azure_cli.set_modules(config=manifest_file,
+                                      connection_string=self.envvars.DEVICE_CONNECTION_INFO.device_id,
+                                      device_id=self.envvars.IOTHUB_CONNECTION_INFO):
+            self.output.footer("DEPLOYMENT COMPLETE")
+
+    def deploy_layered(self, manifest_file: str, layered_deployment_name: str, priority: str):
+
+        self.output.header("DEPLOYING CONFIGURATION")
+
+        self.envvars.verify_envvar_has_val("IOTHUB_CONNECTION_STRING", self.envvars.IOTHUB_CONNECTION_INFO)
+        self.envvars.verify_envvar_has_val("DEPLOYMENT_CONFIG_FILE", self.envvars.DEPLOYMENT_CONFIG_FILE)
+
+        if self.azure_cli.create_layered_deployment(config=manifest_file,
```

Comment:
>manifest_file [](http://example.com/codeflow?start=59&length=13)

if self.envvars.DEPLOYMENT_CONFIG_FILE has value, should we not take this value instead of teh one passed as a parameter?

Reply:
I'd think the parameter overrides the .env, since you'd have to explicitly set it in the command line